tests/AgDatabaseMove.Unit/FileToolsTest.cs

```diff
@@ -0,0 +1,58 @@
+namespace AgDatabaseMove.Unit
+{
+  using System.Collections.Generic;
+  using Xunit;
+
+
+  public class FileToolsTest
+  {
+    public static IEnumerable<object[]> UrlFileExamples => new List<object[]> {
+      new object[] { "https://hello/a.bak" },
+      new object[] { "https://hello/a.full" },
+      new object[] { "https://storage-account.blob.core.windows.net/container/file.trn" },
+      new object[] { "https://hello/a.diff" },
+      new object[] { "https://a.diff" },
+      new object[] { "https://1/2/3/4/5/a.diff" }
+    };
+
+    public static IEnumerable<object[]> NonUrlFileExamples => new List<object[]> {
+      new object[] { @"c:\hello\a.bak" },
+      new object[] { @"\\abc\hello/a.bak" },
+      new object[] { "http://hello/a.bak" },
+      new object[] { "https://storage-account.blob.core.windows.net/container" },
+      new object[] { "https://storage-account.blob.core.windows.net/container/file.bad" }

```

Comment:
Aren't all of these valid URL backup possibilities?

Reply:
None of them are
1. `c:\hello\a.bak` windows path not https
2. `\\abc\hello/a.bak` DFS path not https
3. `http://hello/a.bak` htttp not https
4. `https://storage-account.blob.core.windows.net/container` directory, not a file
5. `https://storage-account.blob.core.windows.net/container/file.bad` invalid extension

Reply:
Sorry, valid in SQL Server. I understand they're valid in this implementation.